drivers/hmis/app/graphql/types/hmis_schema/ce_referral_step.rb

```diff
@@ -8,15 +8,25 @@
 
 module Types
   class HmisSchema::CeReferralStep < Types::BaseObject
-    # object is a Hmis::WorkflowExecution::Step
+    # object is a Hmis::WorkflowExecution::Step, AND also expects context[:referral] is a Hmis::Ce::Referral.
```

Comment:
Flagging this for continued discussion/review

Reply:
Thanks, Let's huddle on this tomorrow!

Without having looked at this too closely, some initial thoughts:
- I would think that we should be able to have a relationship from the step to the referral.  There is a relationship on ce_referrals to the workflow instance
- Assigning to `context` is a new pattern and I'm not sure how it will behave. If for some reason we can't use the relationship, I would suggest reworking this to use an OpenStruct so we can have both objects. `OpenStruct.new(step: step, referral: referral)`. That is typically how we handle the situation where we need multiple unrelated records to implement a resolver (see example: DisabilityGroup resolver).

Reply:
Hi Gig, thanks for looking at it. Sounds good, we can discuss tomorrow. 

To try to summarize my conversation with Theron earlier today, we looked at establishing a direct relationship from Step to Referral; but we were worried that setting up this bidirectional relationship is not best practice. (Bidirectional because Referral already has a reference to Instance, so giving Instance a reference to Referral would be duplicative.) 

But we also did not really want to flip the direction of that existing relationship, since the Referral also wants to know (and currently, does know) about its Instance in order to establish a workflow engine, know what steps are active, etc.

I will pursue your OpenStruct suggestion though, that does sound preferable to messing around with `context`.